Refactoring solve_neos module.

jkawamoto opened the pull request:
Update error handling so that those module do not raise unchecked exception accidentally. To notify errors explicitly, I added NeosError class and raise it when some errors occur during connection between neos server, instead of returning None. I also add noprompt option in neos_solve so that this method does not block batch process.

jkawamoto committed:
Raise ValueError instead of AssertionError when matfile_target doesn't exist in neos_solve.

jkawamoto committed:
Integrate errors raised in processes using selenium to WebDriverException.

jkawamoto committed:
Handle Fault and ProtocolError in NeosInterface.

jkawamoto committed:
Implement NeosError to handle errors caused by neos server.

jkawamoto committed:
Replace a recursive call to a while loop in ask_user_to_submit.

jkawamoto committed:
Add noprompt option in neos_solve so that batch processes do not stop while waiting inputs.

TrishGillett commented on solve_neos.py:
```diff
@@ -50,27 +78,42 @@ def _get_driver():
+
+
+def neos_solve(
+        matfile_target, output_target=None, discard_matfile=True, noprompt=False):
```
Can we change the name noprompt to no_prompt to match the other argument names?

jkawamoto committed:
Rename noprompt in neos_solve to no_prompt.

Following the dicsussion in https://github.com/discardthree/PySDPT3glue/pull/11#discussion_r65823120

jkawamoto commented:
Renamed `noprompt` to `no_prompt` and rebase to follow newest version.
